x-pack/plugins/cases/server/client/user_actions/connectors.ts

```diff
@@ -288,15 +289,15 @@ const createConnectorInfoResult = ({
         latestUserActionDate: latestUserActionCreatedAt,
       });
 
+      const pushDetails = convertEnrichedPushInfoToDetails(enrichedPushInfo);
+
       results[connector.id] = {
         ...connector,
         name: connectorDetails?.name ?? connector.name,
         push: {
           needsToBePushed,
           hasBeenPushed: hasBeenPushed(enrichedPushInfo),
-          externalService: enrichedPushInfo?.externalService,
-          latestUserActionPushDate: enrichedPushInfo?.latestPushDate.toISOString(),
-          oldestUserActionPushDate: enrichedPushInfo?.oldestPushDate.toISOString(),
+          details: pushDetails,
```

Comment:
**nit:** if `details` is optional and `pushDetails` can be `null` wouldn't it be better to do `...(pushDetails && {details: pushDetails})`?

Reply:
Great suggestion 👍